File: stress/t/arr_sieve.c
#include <stdio.h>
#include <stdint.h>
int main(void){
  int64_t n,i,j,count,lastp;
  int t;
  int P[60];
  n=60;
  for(i=0;i<n;i++){ P[i]=1; }
  P[0]=0; P[1]=0;
  for(i=2;i*i<n;i++){
    t=P[i];
    if(t){ for(j=i*i;j<n;j+=i){ P[j]=0; } }
  }
  count=0; lastp=0;
  for(i=0;i<n;i++){ t=P[i]; if(t){ count=count+1; lastp=i; } }
  printf("primes="); printf("%ld",(long)count); printf("\n");
  printf("lastprime="); printf("%ld",(long)lastp); printf("\n");
  t=P[7]; printf("is7prime="); printf("%s", t?"true":"false"); printf("\n");
  t=P[9]; printf("is9prime="); printf("%s", t?"true":"false"); printf("\n");
  return 0;
}

File: stress/t/arr_sieve.f
      PROGRAM ARRSIEVE
      INTEGER*8 N,I,J,COUNT,LASTP
      LOGICAL T
      LOGICAL P(0:59)
      N=60
      I=0
   10 IF (I.GE.N) GOTO 20
      P(I)=.TRUE.
      I=I+1
      GOTO 10
   20 CONTINUE
      P(0)=.FALSE.
      P(1)=.FALSE.
      I=2
   30 IF (I*I.GE.N) GOTO 50
      T=P(I)
      IF (T) THEN
        J=I*I
   40   IF (J.GE.N) GOTO 45
        P(J)=.FALSE.
        J=J+I
        GOTO 40
   45   CONTINUE
      ENDIF
      I=I+1
      GOTO 30
   50 CONTINUE
      COUNT=0
      LASTP=0
      I=0
   60 IF (I.GE.N) GOTO 70
      T=P(I)
      IF (T) THEN
        COUNT=COUNT+1
        LASTP=I
      ENDIF
      I=I+1
      GOTO 60
   70 CONTINUE
      WRITE(*,'(A,I0)') 'primes=',COUNT
      WRITE(*,'(A,I0)') 'lastprime=',LASTP
      T=P(7)
      IF (T) THEN
        WRITE(*,'(A)') 'is7prime=true'
      ELSE
        WRITE(*,'(A)') 'is7prime=false'
      ENDIF
      T=P(9)
      IF (T) THEN
        WRITE(*,'(A)') 'is9prime=true'
      ELSE
        WRITE(*,'(A)') 'is9prime=false'
      ENDIF
      END
